From 0a0f199b3d77d5ee6eab665e03504cdc654aabbd Mon Sep 17 00:00:00 2001
From: Beomho Seo <beomho.seo@samsung.com>
Date: Mon, 12 Aug 2013 21:51:45 +0900
Subject: [PATCH 0527/1302] TRATS2: dts: exynos4412-m0.dts: Add mcs-touchkey
 node 	Support melfas 2 touchkey device for TRATS2 board 	Adds also the
 binding documentation file

Signed-off-by: Beomho Seo <beomho.seo@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 .../devicetree/bindings/input/mcs-touchkey.txt     | 26 ++++++++++++++++++++++
 arch/arm/boot/dts/exynos4412-m0.dts                | 19 ++++++++++++++++
 2 files changed, 45 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/input/mcs-touchkey.txt

diff --git a/Documentation/devicetree/bindings/input/mcs-touchkey.txt b/Documentation/devicetree/bindings/input/mcs-touchkey.txt
new file mode 100644
index 0000000..5454541
--- /dev/null
+++ b/Documentation/devicetree/bindings/input/mcs-touchkey.txt
@@ -0,0 +1,26 @@
+* mcs_touchkey MELFAS MCS5000/5080 controller
+
+Required properties:
+- compatible: must be "mcs5000_touchkey" or "mcs5080_touchkey"
+- reg : I2C address of the chip
+- interrupts: interrupt to which the chop is connected
+- key_maxval: size of keycode table
+- interrupts: interrupt to which the chip is connected
+- code: key code for this device
+
+Example:
+
+	i2c_touch_key: i2c-gpio-0 {
+	       /* ... */
+
+	       touch_key@20 {
+		       compatible = "mcs5080_touchkey";
+		       reg = <0x20>;
+		       interrupt-patrent = <gpj0>;
+		       key_maxval = <2>;
+		       linux, code = <0x0000009e
+			              0x000000a9>;
+	       };
+
+	       /* ... */
+	};
diff --git a/arch/arm/boot/dts/exynos4412-m0.dts b/arch/arm/boot/dts/exynos4412-m0.dts
index 3a43790..fbefee6 100644
--- a/arch/arm/boot/dts/exynos4412-m0.dts
+++ b/arch/arm/boot/dts/exynos4412-m0.dts
@@ -71,6 +71,25 @@
 		};
 	};
 
+	i2c_touch_key: i2c-gpio-3 {
+		compatible = "i2c-gpio";
+		gpios = <&gpl0 2 0>, <&gpl0 1 0>;
+		i2c-gpio,delay-us = <2>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+
+		touch_key@20 {
+			compatible = "mcs5080_touchkey";
+			reg = <0x20>;
+			interrupt-parent = <&gpj0>;
+			interrupts = <3 0>;
+			key_maxval = <2>;
+			linux,code = <0x0000009e	/* KEY_BACK */
+				      0x000000a9>;	/* KEY_MENU */
+		};
+	};
+
 	camera {
 		status = "okay";
 
-- 
1.8.3.2

